### PR TITLE
UART logger

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
- add uart logger
- reduce rate of test task to 10hz
- reduce clutter in build file
- Part 1 of 3 PRs

### Testing Done
Tested manually, logs print on screen

### Resolved Issues

N/A Part of Wifi abstractions, we need the UART logger to look at logs during initialization before we connect to Wifi.
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/firmware_new/boards/frankie_v1/BUILD [hunk 1]
```diff
@@ -78,26 +77,118 @@ cc_library(
...
+        "crc.c",
+    ],
+    hdrs = [
+        "crc.h",
+        "main.h",
+    ],
+    copts = [
         # We don't want to mark warnings as errors for 3rd party code
         "-Wno-error",
+        "-DSTM32H743xx",
```
Comment (jonathanlew) [inline src/firmware_new/boards/frankie_v1/BUILD:115]: what does this do?

File: src/firmware_new/boards/frankie_v1/io/uart_logger.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        log_level = "WARNING";
+    }
+    else if (robot_log.log_level == TbotsProto_LogLevel_FATAL)
+    {
+        log_level = "FATAL";
+    }
+    else
+    {
+        log_level = "UNKOWN";
+    }
```
Comment (jonathanlew) [inline src/firmware_new/boards/frankie_v1/io/uart_logger.c]: can we abstract this into its own function?

File: src/firmware_new/boards/frankie_v1/main.c [hunk 3]
```diff
@@ -168,6 +171,16 @@ int main(void)
     MX_TIM4_Init();
     /* USER CODE BEGIN 2 */
 
+    //               ---- Initialize App/IO Layer ----
+    //
+    // At this point the UART peripheral should be configured correctly,
+    // so we initialize the logger with a UART robot log handler.
+    //
+    // Logs can been seen through `screen /dev/ttyACM0 115200`
+    io_uart_logger_init(&huart3);
```
Comment (jonathanlew) [inline src/firmware_new/boards/frankie_v1/main.c:180]: What does the 3 in huart3 mean?

File: src/firmware_new/boards/frankie_v1/BUILD (hunk 1 above)
Comment (akhilveeraghanta) [reply src/firmware_new/boards/frankie_v1/BUILD:115]: just a macro that you need to set for HAL to work correctly

File: src/firmware_new/boards/frankie_v1/io/uart_logger.c (hunk 2 above)
Comment (akhilveeraghanta) [reply src/firmware_new/boards/frankie_v1/io/uart_logger.c]: done!

File: src/firmware_new/boards/frankie_v1/main.c (hunk 3 above)
Comment (akhilveeraghanta) [reply src/firmware_new/boards/frankie_v1/main.c:180]: its just a name thats assigned by CubeMX, this is uart3 out of ~5 uart peripherals on the H7

File: src/firmware_new/boards/frankie_v1/io/uart_logger.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            return "WARNING";
+        }
+        case TbotsProto_LogLevel_FATAL:
+        {
+            return "FATAL";
+        }
+        default:
+        {
+            return "UNKOWN";
```
Comment (Resocram) [inline src/firmware_new/boards/frankie_v1/io/uart_logger.c]: Misspelled "Unknown"

**Review by Resocram — COMMENTED** [review]
Other than a minor comment, LGTM!

File: src/firmware_new/boards/frankie_v1/io/uart_logger.c (hunk 4 above)
Comment (akhilveeraghanta) [reply src/firmware_new/boards/frankie_v1/io/uart_logger.c]: good catch, fixed!

**Review by jonathanlew — APPROVED** [review]

**Review by Resocram — APPROVED** [review]

**Pull request merged**